src/AutoRest.CSharp/Common/Output/Models/Types/ModelFactoryTypeProvider.cs

```diff
@@ -17,25 +18,27 @@
 using Azure.Core.Expressions.DataFactory;
 using Azure.ResourceManager.Models;
 using Humanizer;
+using Microsoft.CodeAnalysis;
 using static AutoRest.CSharp.Output.Models.MethodSignatureModifiers;
 
 namespace AutoRest.CSharp.Output.Models.Types
 {
-    internal sealed class ModelFactoryTypeProvider : TypeProvider
+    internal sealed class ModelFactoryTypeProvider : SignatureTypeProvider
     {
         protected override string DefaultName { get; }
         protected override string DefaultAccessibility { get; }
 
         private IEnumerable<MethodSignature>? _methods;
-        public IEnumerable<MethodSignature> Methods => _methods ??= Models.Select(CreateMethod);
+        public override IEnumerable<MethodSignature> Methods => _methods ??= Models.Select(CreateMethod).ToList();
```

Comment:
remove `ToList()`? since the return type is `IEnumerable`

Reply:
It is referenced multiple times.
Needs to materialize it here, it throws while visit it the second time.